### DISCUSSION TITLE
blurred background does not work for light themes

### DISCUSSION THREAD

**Discussion by yspreen** [ghostty-org/ghostty#3666] [Feature Requests, Ideas] [answered]
<img width="536" alt="image" src="https://github.com/user-attachments/assets/010e353d-eed5-4602-9fa7-4150a4dbd7d3" />
here's iterm2 on the left vs Ghostty on the right. easy to see that one shows my wallpaper blurred in the background, one doesn't.

if I change the background from ffffff to 000000 it looks much more transparent

<img width="491" alt="image" src="https://github.com/user-attachments/assets/9924c939-3648-4213-98dc-7374d7be1233" />
<img width="688" alt="image" src="https://github.com/user-attachments/assets/06b73596-4737-4248-927c-ba74b509ac0d" />

this is for radius 100 and 10 each. both look exactly the same (1st screenshot) if background is ffffff

config:
```
theme = BlulocoLight
background-blur-radius = 10
background-opacity = 0.5	
background = ffffff
```

**Comment by yspreen** [comment]
I've never been blocked from opening issues on GitHub and banished to the discussion. why gate keep the issues? feels like someone finding a genuine bug gets put behind a wall for no reason

**Reply by 00-kat** [reply]
Please read https://github.com/ghostty-org/ghostty/blob/main/CONTRIBUTING.md and https://github.com/ghostty-org/ghostty/issues/3558.

**Reply by yspreen** [reply]
as a swift developer, I understand this application to use something similar to the swiftui materials which creates a blur effect by increasing saturation & brightness of the background to create the characteristic apple blur effect. those usually come in light and dark varieties and only work correctly when using light materials for light ui and dark material for dark ui.

looking at the source code, using the private APIs is not a perfect solution. I've achieved better blur radius with non-private APIs in apps like my clipboard history which
- works for light and dark mode
- has a configurable blur radius
- allows for a radius that's lower than the ultra thin material

**Comment by yspreen** [comment, marked as answer]
probably #3324
